### ISSUE TITLE
No matplotlib plots are shown when executing notebook

### ISSUE THREAD

**Issue by jaladh-singhal** [jupyter/nbconvert#1081]
Hello team,
Recently I enabled `nbsphinx_execute` in our package [wsynphot](https://github.com/starkit/wsynphot) to execute the notebooks each time docs are built. But I found that matplotlib plots no longer appear in the built docs. Since nbsphinx uses nbconvert, so I tried executing a notebook with `nbconvert --execute` and plots still didn't appear.

I've reproduced this problem by executing a new notebook with minimal plotting code. Here's the produced html for it:
![image](https://user-images.githubusercontent.com/36361463/62414506-0b923380-b63a-11e9-96c1-02621319b08c.png)
which should've been appeared as:
![image](https://user-images.githubusercontent.com/36361463/62414585-0aadd180-b63b-11e9-8f1d-a7a77e0611cf.png)

I also found that when producing notebook i.e. `nbconvert --execute --to notebook`, the plot still doesn't appear same as in case of html.

**Nbconvert version: 5.5.0**

**Comment by MSeal** [comment]
So this is more of an ipython or ipywidgets topic. This has to do with the `notebook` format returning javascript that needs a browser to render correctly. See https://nbconvert.readthedocs.io/en/latest/execute_api.html#widget-state. Normally this is fine and the widgets will save state the allows us to render the outcome for html / notebooks (https://github.com/jupyter/nbconvert/pull/900 added the capability here). Trying 
```
import ipywidgets as widgets
from IPython.display import display
w = widgets.IntSlider()
display(w)
```
will show that this renders a slider at default locations. I looked at the notebook output and the issue is that this particular widget is not returning `"application/vnd.jupyter.widget-view+json"` data output type. This means it's relying 100% on a browser to render the results rather than supplying the associated widget information for other systems to render. I'd open an issue in the appropriate project and point to these notes for why it's not rendering.

Alternatively in the meanwhile if you change to `inline` for the matplot lib rendering it renders as expected since this doesn't generate a javascript version of the plot.

**Comment by WhySeeYC** [comment]
I have a similar issue but not entirely the same. Hope it is ok to share it here as well. 
I tried to run `jupyter nbconvert mynotebook.ipynb --no-input --to html` and the html file was successfully created but the annotations, labels in the graphs are not properly shown.
For example, the following is a graph shown in browser after opening the html file:
<img width="952" alt="image" src="https://github.com/jupyter/nbconvert/assets/59447026/64ab6914-2d11-4b88-b36f-d6d3e765ad6d">

But in actual working notebook, it should look like the following:
(Apologise that it is only partial of the image, I wasn't able to screenshot any larger area)
<img width="1057" alt="image" src="https://github.com/jupyter/nbconvert/assets/59447026/5ccc2436-874c-42f8-a8d6-61ce92c9fa2e">


In other words, only part of the graph is being rendered in the HTML file (e.g. the histograms) but some were left out (the scatter plots). 
It will be very appreciated for some guidance 🙏🏻
